gammapy/spectrum/spectrum_fit.py

```diff
@@ -293,12 +290,13 @@ def _run_sherpa_fit(self):
         thres_lo = self.energy_threshold_low.to('keV').value
         thres_hi = self.energy_threshold_high.to('keV').value
 
+        namedataset = []
         for i in range(len(ds.datasets)):
             datastack.notice_id(i + 1, thres_lo[i], thres_hi[i])
-
+            namedataset.append(i + 1)
         datastack.set_stat(self.statistic)
-        ds.fit()
-        datastack.covar()
+        ds.fit(*namedataset)
```

Comment:
I encountered some problems with sherpa when in the same script you fit multiples times. For example, when I tested the grouping I fit multiple times for different groups and sherpa if you don't add this option keep in memory the others dataset so it gives you each time the result for the different datasets that you used in your script. This line allows to force sherpa during the fit to take only into account the dataset that you gives with the ogip files. I'm not sure if this is clear...